_studio/mfx_lib/shared/src/mfx_mpeg2_encode_vaapi.cpp

```diff
@@ -370,24 +371,21 @@ mfxStatus VAAPIEncoder::QueryEncodeCaps(ENCODE_CAPS & caps, mfxU8 codecProfileTy
         MFX_CHECK_STS(mfxSts);
     }
 
-    // m_width  = width;
-    // m_height = height;
+    memset(&m_caps, 0, sizeof(m_caps));
 
-    memset(&caps, 0, sizeof(caps));
+    m_caps.EncodeFunc        = 1;
+    m_caps.BRCReset          = 1; // No bitrate resolution control
+    m_caps.VCMBitrateControl = 0; // Video conference mode
+    m_caps.HeaderInsertion   = 0; // We will provide headers (SPS, PPS) in binary format to the driver
+    m_caps.MbQpDataSupport   = 1;
+    m_caps.SkipFrame         = 1;
 
-    caps.EncodeFunc        = 1;
-    caps.BRCReset          = 1; // No bitrate resolution control
-    caps.VCMBitrateControl = 0; // Video conference mode
-    caps.HeaderInsertion   = 0; // We will provide headers (SPS, PPS) in binary format to the driver
-    caps.MbQpDataSupport   = 1;
-    caps.SkipFrame         = 1;
-
-    caps.SliceIPBOnly      = 1;
-    caps.MaxNum_Reference  = 1;
-    caps.MaxPicWidth       = 1920;
-    caps.MaxPicHeight      = 1088;
-    caps.NoInterlacedField = 0; // Enable interlaced encoding
-    caps.SliceStructure    = 1; // 1 - SliceDividerSnb; 2 - SliceDividerHsw; 3 - SliceDividerBluRay; the other - SliceDividerOneSlice
+    m_caps.SliceIPBOnly      = 1;
+    m_caps.MaxNum_Reference  = 1;
+    m_caps.MaxPicWidth       = 1920;
+    m_caps.MaxPicHeight      = 1088;
+    m_caps.NoInterlacedField = 0; // Enable interlaced encoding
+    m_caps.SliceStructure    = 1; // 1 - SliceDividerSnb; 2 - SliceDividerHsw; 3 - SliceDividerBluRay; the other - SliceDividerOneSlice
```

Comment:
Why we don't have enum for different slice structures. @dmitryermilov : I think we touched something similar to that recently... Maybe that's time to fix that?

Reply:
https://github.com/Intel-Media-SDK/MediaSDK/pull/1814

Reply:
commit wasn't merged